src/pinecone.ts

```diff
@@ -320,7 +311,14 @@ export class Pinecone {
    * @throws {@link Errors.PineconeArgumentError} when invalid arguments are provided
    */
   deleteIndex(indexName: IndexName) {
-    return this._deleteIndex(indexName);
+    const deleteIndexPromise = this._deleteIndex(indexName);
+
+    // When an index is deleted, we need to evict the host from the IndexHostSingleton cache.
+    deleteIndexPromise.then(() => {
```

Comment:
I believe this should work for evicting things from the cache if the index has been deleted.